File: lib/models/testimonial.dart
class Testimonial {
  final int? id;
  final String studentName;
  final String message;
  final String examName;
  final String rank;
  final int? rating;

  Testimonial({
    this.id,
    required this.studentName,
    required this.message,
    required this.examName,
    this.rank = '',
    this.rating,
  });

  factory Testimonial.fromJson(Map<String, dynamic> json) {
    return Testimonial(
      id: json['id'] as int?,
      studentName: json['student_name'] as String? ?? '',
      message: json['message'] as String? ?? '',
      examName: json['exam_name'] as String? ?? '',
      rank: json['rank'] as String? ?? '',
      rating: json['rating'] as int?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'student_name': studentName,
      'message': message,
      'exam_name': examName,
      'rank': rank,
      'rating': rating,
    };
  }
}

File: backend/app/models/testimonial.py
from sqlalchemy import Column, Integer, String, DateTime, Text
from datetime import datetime, timezone
from ..database import Base


class Testimonial(Base):
    __tablename__ = "testimonials"

    id = Column(Integer, primary_key=True, index=True)
    student_name = Column(String, nullable=False)
    message = Column(Text, nullable=False)
    exam_name = Column(String, default="")
    rank = Column(String, default="")
    rating = Column(Integer, default=5)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))
